### ISSUE TITLE
statefulset is not ready

### ISSUE THREAD

**Issue by Chenhb123** [karmada-io/karmada#1193]
**Please provide an in-depth description of the question you have**:
I tried to deploy a statefulset to the specified cluster named **kubernetes-admin144**, After completing the deployment, I found that
statefulset is not ready, However, the statefulset was actually successful deployed in **kubernetes-admin144**, Here's how I did it:
```bash
[karmada/samples/nginx]# kubectl config use-context karmada-apiserver
Switched to context "karmada-apiserver".
[karmada/samples/nginx]# cat nginx-statefulset.yaml
apiVersion: v1
kind: Service
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  ports:
  - port: 80
    name: web
  clusterIP: None
  selector:
    app: nginx
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: web
spec:
  selector:
    matchLabels:
      app: nginx # has to match .spec.template.metadata.labels
  serviceName: "nginx"
  replicas: 2 # by default is 1
  template:
    metadata:
      labels:
        app: nginx # has to match .spec.selector.matchLabels
    spec:
      terminationGracePeriodSeconds: 10
      containers:
      - name: nginx
        image: registry.aliyuncs.com/google_containers/nginx-slim:0.8
        ports:
        - containerPort: 80
          name: web
        volumeMounts:
        - name: www
          mountPath: /usr/share/nginx/html
  volumeClaimTemplates:
  - metadata:
      name: www
    spec:
      accessModes: [ "ReadWriteOnce" ]
      storageClassName: "nfs-client"
      resources:
        requests:
          storage: 1Gi
[karmada/samples/nginx]# cat nginx-statefulset-propagation.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: nginx-web
spec:
  resourceSelectors:
    - apiVersion: v1
      kind: Service
      name: nginx
    - apiVersion: apps/v1
      kind: StatefulSet
      name: web
  placement:
    clusterAffinity:
      clusterNames:
        - kubernetes-admin144
 
[karmada/samples/nginx]# kubectl apply -f nginx-statefulset.yaml -f nginx-statefulset-propagation.yaml
service/nginx created
statefulset.apps/web created
propagationpolicy.policy.karmada.io/nginx-web created

[karmada/samples/nginx]# kubectl get svc
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   2d1h
nginx        ClusterIP   None         <none>        80/TCP    2m15s
[karmada/samples/nginx]# kubectl get statefulset
NAME   READY   AGE
web    0/2     2m21s
[karmada/samples/nginx]#
[karmada/samples/nginx]# kubectl --kubeconfig=/root/.kube/144.config get svc
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.1.0.1     <none>        443/TCP   3d
nginx        ClusterIP   None         <none>        80/TCP    2m40s
[karmada/samples/nginx]# kubectl --kubeconfig=/root/.kube/144.config get statefulset
NAME   READY   AGE
web    2/2     2m44s
[karmada/samples/nginx]# kubectl --kubeconfig=/root/.kube/144.config get po
NAME                                      READY   STATUS    RESTARTS   AGE
nfs-client-provisioner-788786ff4d-5wznv   1/1     Running   0          19h
web-0                                     1/1     Running   0          2m48s
web-1                                     1/1     Running   0          2m45s
[karmada/samples/nginx]#
```
**What do you think about this question?**:
I don't know how to check this problem, Where can I view logs or is there a problem with my yaml file. The last, I hope to know the cause of this problem,And how to solve it.
**Environment**:
- Karmada version:
```bash
[karmada/samples/nginx]# karmadactl version
karmadactl version: version.Info{GitVersion:"v0.0.0-master", GitCommit:"unknown", GitTreeState:"unknown", BuildDate:"unknown", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version:
```bash
[karmada/samples/nginx]# kubectl get clusters
NAME                  VERSION   MODE   READY   AGE
karmada-apiserver     v1.21.1   Pull   True    2d1h
kubernetes-admin144   v1.17.4   Push   True    19h
localhost             v1.17.4   Push   True    20h
member1               v1.21.1   Push   True    2d1h
member2               v1.21.1   Push   True    2d1h
member3               v1.21.1   Push   True    2d
```
- Others:

**Comment by lonelyCZ** [comment]
It seem not to support `statefulset` AggregateStatus now.

https://github.com/karmada-io/karmada/blob/075bfc97ba9d66de568e6642d652ec5e75398b53/pkg/detector/detector.go#L1029-L1042

**Comment by Chenhb123** [comment]
> It seem not to support `statefulset` AggregateStatus now.
> 
> https://github.com/karmada-io/karmada/blob/075bfc97ba9d66de568e6642d652ec5e75398b53/pkg/detector/detector.go#L1029-L1042

emmmm, 👌Ok, thank you very much for your answer

**Comment by Chenhb123** [comment]
Are there plans to add statefulset AggregateStatus to project development in the future, If so, when will this feature be added. Looking forward to your reply. @lonelyCZ

**Comment by RainbowMango** [comment]
@Chenhb123 It's part of the [resource-interpreter-webhook](https://github.com/karmada-io/karmada/tree/master/docs/proposals/resource-interpreter-webhook), and what you are asking is `InterpretStatus` and `AggregateStatus` operations. 

We have implemented `Retain`/`InterpretReplica`/`ReviseReplica`, I think the  `InterpretStatus` and `AggregateStatus` will be the next.

We can't guarantee but I guess it will be included in v1.1, it depends on if we have enough community effort.

**Comment by Chenhb123** [comment]
> @Chenhb123 It's part of the [resource-interpreter-webhook](https://github.com/karmada-io/karmada/tree/master/docs/proposals/resource-interpreter-webhook), and what you are asking is `InterpretStatus` and `AggregateStatus` operations.
> 
> We have implemented `Retain`/`InterpretReplica`/`ReviseReplica`, I think the `InterpretStatus` and `AggregateStatus` will be the next.
> 
> We can't guarantee but I guess it will be included in v1.1, it depends on if we have enough community effort.

I see😬,Thanks again for your reply.
